pkg/pillar/hypervisor/xen.go

```diff
@@ -402,45 +402,14 @@ func (ctx xenContext) CreateDomConfig(domainName string, config types.DomainConf
 }
 
 func (ctx xenContext) Stop(domainName string, domainID int, force bool) error {
-	log.Infof("xlShutdown %s %d\n", domainName, domainID)
-	args := []string{
-		"xl",
-		"shutdown",
-		domainName,
-	}
-	if force {
-		args = append(args, "-F")
-	}
-	ctrdCtx, done := ctx.ctrdClient.CtrNewUserServicesCtx()
-	defer done()
-	stdOut, stdErr, err := ctx.ctrdClient.CtrExec(ctrdCtx, domainName, args)
-	if err != nil {
-		log.Errorln("xl shutdown failed ", err)
-		log.Errorln("xl shutdown output ", stdOut, stdErr)
-		return fmt.Errorf("xl shutdown failed: %s %s", stdOut, stdErr)
-	}
-	log.Infof("xl shutdown done\n")
-	return nil
+	return ctx.ctrdContext.Stop(domainName, domainID, true)
```

Comment:
I see that even for KVM Stop() we have code before calling this function. Are you sure we don't need something for Xen?

And this should be graceful., hence force argument should be false, right?

Reply:
I don't think this is correct @zed-rishabh -- this will result in a task (from containerd perspective) being killed pretty violently.

Now, you're right thinking that down the road we will be able to use that mechanism -- but that would require a bit more sophisticated signal handler on xen-init side.

In short -- I don't think we should be doing it this way BEFORE we fully transition into containerd workflow with signals, etc.